### PR TITLE
paddle dropout function

### LINKED ISSUE #17415: dropout

### PR DESCRIPTION
**Pull request by bsameer65**
Close #17415

### CODE REVIEW

**Comment by Daniel4078** [comment]
add_frontend_checklist

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello! I just checked that your test is currently passing fine. But you do still need to fix some formatting errors. Check my comments below for some details. Thanks!

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello! I just checked that your test is currently passing fine. But you do still need to fix some formatting errors. Check my comments below for some details. Thanks!

**Comment by bsameer65** [comment]
Hi, can you review it again? Thankyou.

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
 
     cosine = numerator / denominator
     return cosine
+
+def get_mask(shape,device,prob,seed=None):
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:27]: -def get_mask(shape,device,prob,seed=None):
+
+def get_mask(shape, device, prob, seed=None):

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 2]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+    mask = ivy.where(
+        ivy.random_uniform(shape=shape, device=device, seed=seed)
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:29]: -        ivy.random_uniform(shape=shape, device=device, seed=seed)
-        < prob,
+        ivy.random_uniform(shape=shape, device=device, seed=seed) < prob,

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 3]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+        < prob,
+        0.0,
+        1.0,
+    )
+    return mask
+
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:35]: one extra empty line needed here

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 4]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+@with_supported_dtypes({"2.4.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def dropout(x, p=0.5, axis=None, training=True,
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:38]: -def dropout(x, p=0.5, axis=None, training=True,
-            mode='upscale_in_train', name=None):
-    
+def dropout(x, p=0.5, axis=None, training=True, mode="upscale_in_train", name=None):

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 5]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+            mode='upscale_in_train', name=None):
+    
+    if axis > 1:
+        raise ValueError("Axis value can only be 0 or 1 or None.")
+
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:43]:
```
-        
-    elif axis==None or (isinstance(axis,list) and len(axis==2)):
-        mask = get_mask(shape=x.shape, device=ivy.dev(x),prob=p, seed=None)
-    elif axis==0:
-        mask = get_mask(shape=(x.shape[0],1), device=ivy.dev(x), prob=p)
+
+    elif axis == None or (isinstance(axis, list) and len(axis == 2)):
+        mask = get_mask(shape=x.shape, device=ivy.dev(x), prob=p, seed=None)
+    elif axis == 0:
+        mask = get_mask(shape=(x.shape[0], 1), device=ivy.dev(x), prob=p)
```

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 6]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+    
+    if axis > 1:
+        raise ValueError("Axis value can only be 0 or 1 or None.")
+        
+    elif axis==None or (isinstance(axis,list) and len(axis==2)):
+        mask = get_mask(shape=x.shape, device=ivy.dev(x),prob=p, seed=None)
+    elif axis==0:
+        mask = get_mask(shape=(x.shape[0],1), device=ivy.dev(x), prob=p)
+        mask = ivy.broadcast_to(mask, x.shape)
+    elif axis==1:
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:49]:
```
-    elif axis==1:
-        mask = get_mask(shape=(1,x.shape[1]), device=ivy.dev(x), prob=p)
+    elif axis == 1:
+        mask = get_mask(shape=(1, x.shape[1]), device=ivy.dev(x), prob=p)
```

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 7]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+        mask = get_mask(shape=(1,x.shape[1]), device=ivy.dev(x), prob=p)
+        mask = ivy.broadcast_to(mask, x.shape)
+
+    if mode == 'upscale_in_train':
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:53]:
```
-    if mode == 'upscale_in_train':
+    if mode == "upscale_in_train":
```

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 8]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+        if training:
+            out = ivy.multiply(x, mask)
+            ret= ivy.multiply(out, 1.0 / (1.0 - p))
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:56]:
```
-            ret= ivy.multiply(out, 1.0 / (1.0 - p))
+            ret = ivy.multiply(out, 1.0 / (1.0 - p))
```

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 9]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+        else:
+            ret = x
+    else:
+        if training:
+            ret = ivy.multiply(x, mask)
+        else:
+            ret = ivy.multiply(x,(1.0-p))
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:63]:
```
-            ret = ivy.multiply(x,(1.0-p))
-    return ret
+            ret = ivy.multiply(x, (1.0 - p))
+    return ret
```

File: ivy/functional/frontends/paddle/nn/functional/common.py [hunk 10]
```diff
@@ -23,3 +23,42 @@ def cosine_similarity(x1, x2, *, axis=1, eps=1e-08):
...
+    return ret
```
Comment (Daniel4078) [inline ivy/functional/frontends/paddle/nn/functional/common.py:64]: please keep an empty line at the end of files

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_common.py [hunk 11]
```diff
@@ -41,3 +41,47 @@ def test_paddle_cosine_similarity(
         x2=x[1],
         axis=axis,
     )
+
+#dropout
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_common.py:45]: -#dropout
+
+# dropout

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_common.py [hunk 12]
```diff
@@ -41,3 +41,47 @@ def test_paddle_cosine_similarity(
...
+        num_arrays=1,
+        shared_dtype=True,
+        min_value=2,
+        max_value=5,
+        min_num_dims=1,
+        max_num_dims=1,
+        min_dim_size=2,
+    ),
+    p=st.floats(min_value=0.0, max_value=1.0),
+    axis= st.integers(min_value=0, max_value=1),
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_common.py:59]:
```
-    axis= st.integers(min_value=0, max_value=1),
-    training= st.booleans() ,
-    mode= st.one_of(*[st.just(seq) for seq in ['upscale_in_train','downscale_in_infer']]),
+    axis=st.integers(min_value=0, max_value=1),
+    training=st.booleans(),
+    mode=st.one_of(
+        *[st.just(seq) for seq in ["upscale_in_train", "downscale_in_infer"]]
+    ),
```

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_common.py [hunk 13]
```diff
@@ -41,3 +41,47 @@ def test_paddle_cosine_similarity(
...
+        p=p,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        test_flags=test_flags,
+        on_device=on_device,
+        x=x[0],
+        training=training,
+        axis=axis,
+        mode=mode,
+    )
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_common.py:87]: please keep an empty line at the end of files

**Review by Daniel4078 — APPROVED** [review]
Looks good to me!

**Pull request merged**